frontend/src/components/visual-editor/form/inputs/message/QuickRepliesInput.tsx

```diff
@@ -47,16 +48,11 @@ const QuickRepliesInput: FC<QuickRepliesInput> = ({
     const updatedQuickReplies = [...quickReplies];
 
     updatedQuickReplies.splice(index, 1);
+    // Set updated state only if it's greater than minInput
     setQuickReplies(
-      updatedQuickReplies.length
+      updatedQuickReplies.length >= minInput
         ? updatedQuickReplies
-        : [
-            createValueWithId({
-              content_type: QuickReplyType.text,
-              title: "",
-              payload: "",
-            }),
-          ],
+        : updatedQuickReplies.length ? updatedQuickReplies : []
```

Comment:
`setQuickReplies(updatedQuickReplies)` is entirely sufficient; the condition on updatedQuickReplies.length isn't necessary here.
If `length=0` than the array is already an empty array, and if it's less than `minInput` we actually need to set to to the new value not to an empty array.

The source of the problem was the new object 
```
[
            createValueWithId({
              content_type: QuickReplyType.text,
              title: "",
              payload: "",
            }),
          ]
```
Great job identifying that!